libraries/AP_RangeFinder/RangeFinder.cpp

```diff
@@ -680,8 +681,14 @@ void RangeFinder::detect_instance(uint8_t instance, uint8_t& serial_instance)
     case RangeFinder_TYPE_VL53L0X:
         if (!_add_backend(AP_RangeFinder_VL53L0X::detect(state[instance],
                                                          hal.i2c_mgr->get_device(1, 0x29)))) {
-            _add_backend(AP_RangeFinder_VL53L0X::detect(state[instance],
-                                                        hal.i2c_mgr->get_device(0, 0x29)));
+            if (!_add_backend(AP_RangeFinder_VL53L0X::detect(state[instance],
+                                                             hal.i2c_mgr->get_device(0, 0x29)))) {
```

Comment:
Hi @muramura,

Thank you so much for reviewing my code, and help me test this sensor!

- The **I2C address** can be changed, but it will be reset to default after power cycle so I don't think it will be useful to change the address.
- I'm thinking of adding **range** mode parameter too, but we need to consider: range setting should only be changed before flight, and we also need update RFND_MAX_DISTANCE parameter too. I will try it in this weekend and let you know!

Please tell me if you find any issue while testing with your hardware, I have just updated my code. Thanks!